res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1459 @@
+    - Hilton Rudham: Pioneer DDJ-SR mapping
+      https://github.com/hrudham/Mixxx-Pioneer-DDJ-SR
+    - Jan Holthuis: Roland DJ-505 mapping
+	  original brief: https://mixxx.discourse.group/t/roland-dj-505/17916 but now part of main.
+
+    This program is free software; you can redistribute it and/or modify it under the terms of the
+    GNU General Public License as published by the Free Software Foundation; either version 2
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+    without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See
+    the GNU General Public License for more details.
+    
+    You should have received a copy of the GNU General Public License along with this program; if
+    not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+*/
+// eslint-disable-next-line no-var
+var DDJSR2 = {};
+
+///////////////////////////////////////////////////////////////
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
+
+// Sets the jogwheels sensitivity. 1 is default, 2 is twice as sensitive, 0.5 is half as sensitive.
+DDJSR2.jogwheelSensitivity = 2;
+
+// Sets how much more sensitive the jogwheels get when holding shift.
+// Set to 1 to disable jogwheel sensitivity increase when holding shift (default: 10).
+DDJSR2.jogwheelShiftMultiplier = 10;
+
+// If true, vu meters twinkle if AutoDJ is enabled (default: true).
+DDJSR2.twinkleVumeterAutodjOn = true;
+// If true, selected track will be added to AutoDJ queue-top on pressing shift + rotary selector,
+// else track will be added to AutoDJ queue-bottom (default: false).
+DDJSR2.autoDJAddTop = false;
+// Sets the duration of sleeping between AutoDJ actions if AutoDJ is enabled [ms] (default: 1000).
+DDJSR2.autoDJTickInterval = 1000;
+// Sets the maximum adjustment of BPM allowed for beats to sync if AutoDJ is enabled [BPM] (default: 10).
+DDJSR2.autoDJMaxBpmAdjustment = 10;
+// If true, AutoDJ queue is being shuffled after skipping a track (default: false).
+// When using a fixed set of tracks without manual intervention, some tracks may be unreachable,
+// due to having an unfortunate place in the queue ordering. This solves the issue.
+DDJSR2.autoDJShuffleAfterSkip = false;
+
+// If true, by releasing rotary selector,
+// track in preview player jumps forward to "jumpPreviewPosition"
+// (default: jumpPreviewEnabled = true, jumpPreviewPosition = 0.3).
+DDJSR2.jumpPreviewEnabled = true;
+DDJSR2.jumpPreviewPosition = 0.3;
+
+
+// DDJSR2.shiftPressed = false;
+// DDJSR2.shiftPanelSelectPressed = false;
+
+DDJSR2.rateRanges = [0.08, 0.16, 0.32, 0.64, 0.90, 1.00];
+
+DDJSR2.scratchSettings = {
+    "alpha": 1.0 / 8,
+    "beta": 1.0 / 8 / 32,
+    "wheelResolution": 20000,
```

Comment:
> Needs calibration. 50000 seems to be correct, where a 360 rotation of the wheel is the same as a 360 rotation in the Spinning Vinyl onscreen. Is this not the intended purpose?

Yes that is usually the intended purpose. A one-to-one mapping between the onscreen vinyl sticker and the physical one (in theory if there was one). I'm just confused because usually the number if ticks per rotation is much lower (~1000 on highend controllers) so either the calculations in the mapping are wrong or pioneer is cheating again because I don't think their jogwheel is actually that high-res.